backend/src/controllers/deck.js

```diff
@@ -0,0 +1,172 @@
+import deckService from '../services/deck.js';
+import userService from '../services/user.js';
+import cardService from '../services/card.js';
+
+import { Card } from '../db/index.js';
+
+export default {
+  /**
+   * Expresse.js controller for GET /decks
+   * @param {import('express').Request} req
+   * @param {import('express').Response} res
+   * @param {import('express').NextFunction} next
+   * @returns {Promise<void>}
+   */
+  cget: async (req, res, next) => {
+    try {
+      const decks = await deckService.findAll();
+      res.json(decks);
+    } catch (err) {
+      next(err);
+    }
+  },
+  /**
+   * Express.js controller for POST /deck
+   * @param {import('express').Request} req
+   * @param {import('express').Response} res
+   * @param {import('express').NextFunction} next
+   * @returns {Promise<void>}
+   */
+  post: async (req, res, next) => {
+    try {
+      const deck = await deckService.create({
+        name: req.body.name,
+        userId: req.user.id,
+      });
+      const createdDeck = await deckService.findById(deck.id);
+      res.status(201).json(createdDeck);
+    } catch (err) {
+      next(err);
+    }
+  },
+  /**
+   * Express.js controller for GET /decks/:id
+   * @param {import('express').Request} req
+   * @param {import('express').Response} res
+   * @param {import('express').NextFunction} next
+   * @returns {Promise<void>}
+   * */
+  get: async (req, res, next) => {
+    try {
+      const deck = await deckService.findById(parseInt(req.params.id), {
+        include: Card,
+      });
+      if (!deck) return res.sendStatus(404);
+
+      res.json(deck);
+    } catch (err) {
+      next(err);
+    }
+  },
+  /**
+   * Express.js controller for patch /decks/:id
+   * @param {import('express').Request} req
+   * @param {import('express').Response} res
+   * @param {import('express').NextFunction} next
+   * @returns {Promise<void>}
+   */
+  patch : async (req, res, next) => {
+    try {
+      const [deck] = await deckService.update(
+        { id: parseInt(req.params.id) },
+        req.body,
```

Comment:
Fait attention aux injections ici. Si le user trouve le nom d'une column de ta base, il peut injecter ce qu'il veut.
```suggestion
        {
          myColumn: body.myColumn,
        },
```